src/main/java/dev/blaauwendraad/masker/json/ValueMaskers.java

```diff
@@ -112,11 +112,38 @@ public static ValueMasker.NumberMasker eachDigitWith(int digit) {
         }
         byte[] replacementBytes = String.valueOf(digit).getBytes(StandardCharsets.UTF_8);
         return describe(
-                "every digit as %s".formatted(digit),
+                "every digit as integer: %s".formatted(digit),
                 context -> context.replaceBytes(0, context.byteLength(), replacementBytes, context.byteLength())
         );
     }
 
+    /**
+     * Masks all digits of a target numeric value with a static String (which can also be a single
+     * character).
+     *
+     * <p> For example, {@literal "maskMe": 12345 -> "maskMe": "*****"}.
+     * <p> Or, for example {@literal "maskMe": 123 -> "maskMe": "NoNoNo"}.
+     */
+    public static ValueMasker.NumberMasker eachDigitWith(String value) {
+        return describe(
+                "every digit as string: %s".formatted(value),
+                context -> {
+                    byte[] maskValueBytes = value.getBytes(StandardCharsets.UTF_8);
```

Comment:
Do this outside of lambda so it is caches for all masked values

Reply:
Nice catch! I moved both lines to outside of the lambda